lib/zold/commands/node.rb

```diff
@@ -157,7 +158,7 @@ def run(args = [])
       Front.set(:reboot, !opts['never-reboot'])
       invoice = opts[:invoice]
       unless invoice.include?('@')
-        if @wallets.find(Id.new(invoice)).exists?
+        if HungryWallets.new(@wallets).find(Id.new(invoice)).exists?
```

Comment:
@botanicus hm.. I didn't understand. Why we use it only here? What about other places, where `@wallets` are used? We should decorate the object once and pass it everywhere later, both to `Entrance`, `Front`, etc.

Reply:
So taking `Node` as the entry point, from there on it goes everywhere, but it’s _only_ that, `Node` and wherever `Node` passes it to, correct? Aka it’s _not_ a global replace that every instance of `Wallets` should in fact be `HungryWallets`.

Correct?

Reply:
Yes, correct. We only need HungryWallets inside a node. In all other places, it's pointless to do this "hungry pulling" operation.